### PR TITLE
Lua: conan v2 support & v5.3.6

### PR DESCRIPTION
**Pull request by StellaSmith**

### CODE REVIEW

**Comment by StellaSmith** [comment]
Huh. The command line argument didn't get escaped? Is this intended behavior?

File: recipes/lua/all/conanfile.py [hunk 1]
```diff
@@ -1,72 +1,78 @@
-from conans import ConanFile, CMake, tools
 import os
 
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+from conan.tools.files import get, copy, load, save, export_conandata_patches, apply_conandata_patches, collect_libs
+
+
 required_conan_version = ">=1.33.0"
```
Comment (uilianries) [inline recipes/lua/all/conanfile.py:8]:
```suggestion
required_conan_version = ">=1.53.0"
```
self.settings.rm_safe requires Conan 1.53

File: recipes/lua/all/test_v1_package/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+find_package(lua REQUIRED CONFIG)
+
+add_executable(${PROJECT_NAME} test_package.cpp)
+target_link_libraries(${PROJECT_NAME} lua::lua)
+if(COMPILE_AS_CPP)
+  target_compile_definitions(${PROJECT_NAME} PRIVATE COMPILE_AS_CPP)
+endif()
```
Comment (uilianries) [inline recipes/lua/all/test_v1_package/CMakeLists.txt:13]:
```suggestion
add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../test_package/
                 ${CMAKE_CURRENT_BINARY_DIR}/test_package/)
```
So we can re-use from test_package/Cmakelists.txt

File: recipes/lua/all/test_v1_package/test_package.cpp [hunk 3]
```diff
@@ -0,0 +1,20 @@
+
```
Comment (uilianries) [inline recipes/lua/all/test_v1_package/test_package.cpp:1]: This file can be deleted, as we re-use it from test_package/ folder instead.

**Review by uilianries — COMMENTED** [review]
Thank you for updating it. Just a small change on test package to re-use same files, so we don't need to update more files. That suggestiong is already integrated to CCI cmake template: https://github.com/conan-io/conan-center-index/tree/master/docs/package_templates/cmake_package/all

**Review by uilianries — DISMISSED** [review]
LGTM

File: recipes/lua/all/test_v1_package/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,7 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../test_package/ ${CMAKE_CURRENT_BINARY_DIR}/test_package/)
```
Comment (uilianries) [inline recipes/lua/all/test_v1_package/CMakeLists.txt:7]:
```suggestion
add_subdirectory(${CMAKE_CURRENT_SOURCE_DIR}/../test_package/ ${CMAKE_CURRENT_BINARY_DIR}/test_package/)

```
Hooks does not accept files without an empty line

Comment (StellaSmith) [reply recipes/lua/all/test_v1_package/CMakeLists.txt]: Good to know

**Review by uilianries — DISMISSED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Review by RubenRBS — APPROVED** [review]
lgtm!

**Pull request merged**